Large zip files fail to scan

#54 

```
[ERROR] FileNotFoundError: [Errno 2] No such file or directory: '7za'
Traceback (most recent call last):
  File "/var/lang/lib/python3.8/site-packages/aws_lambda_powertools/metrics/metrics.py", line 184, in decorate
    response = lambda_handler(event, context)
  File "/var/lang/lib/python3.8/site-packages/aws_lambda_powertools/logging/logger.py", line 347, in decorate
    return lambda_handler(event, context)
  File "/var/task/lambda.py", line 76, in lambda_handler
    expand_if_large_archive(
  File "/var/task/lambda.py", line 142, in expand_if_large_archive
    archive_summary = subprocess.run(
  File "/var/lang/lib/python3.8/subprocess.py", line 493, in run
    with Popen(*popenargs, **kwargs) as process:
  File "/var/lang/lib/python3.8/subprocess.py", line 858, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "/var/lang/lib/python3.8/subprocess.py", line 1704, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
```


[This function](https://github.com/awslabs/cdk-serverless-clamscan/blob/main/assets/lambda/code/scan/lambda.py#L123) is not properly handling large ZIP files.

--- a/API.md
+++ b/API.md
@@ -40,7 +40,7 @@ Successful Scan Event format
     "source": "serverless-clamscan",
     "input_bucket": <input_bucket_name>,
     "input_key": <object_key>,
-    "status": <"CLEAN"|"INFECTED">,
+    "status": <"CLEAN"|"INFECTED"|"N/A">,
     "message": <scan_summary>,
   }
 ```
@@ -68,7 +68,7 @@ new ServerlessClamscan(scope: Construct, id: string, props: ServerlessClamscanPr
   * **defsBucketAccessLogsConfig** (<code>[ServerlessClamscanLoggingProps](#cdk-serverless-clamscan-serverlessclamscanloggingprops)</code>)  Whether or not to enable Access Logging for the Virus Definitions bucket, you can specify an existing bucket and prefix (Default: Creates a new S3 Bucket for access logs ). __*Optional*__
   * **efsEncryption** (<code>boolean</code>)  Whether or not to enable encryption on EFS filesystem (Default: enabled). __*Optional*__
   * **onError** (<code>[IDestination](#aws-cdk-aws-lambda-idestination)</code>)  The Lambda Destination for files that fail to scan and are marked 'ERROR' or stuck 'IN PROGRESS' due to a Lambda timeout (Default: Creates and publishes to a new SQS queue if unspecified). __*Optional*__
-  * **onResult** (<code>[IDestination](#aws-cdk-aws-lambda-idestination)</code>)  The Lambda Destination for files marked 'CLEAN' or 'INFECTED' based on the ClamAV Virus scan (Default: Creates and publishes to a new Event Bridge Bus if unspecified). __*Optional*__
+  * **onResult** (<code>[IDestination](#aws-cdk-aws-lambda-idestination)</code>)  The Lambda Destination for files marked 'CLEAN' or 'INFECTED' based on the ClamAV Virus scan or 'N/A' for scans triggered by S3 folder creation events marked (Default: Creates and publishes to a new Event Bridge Bus if unspecified). __*Optional*__
 
 
 
@@ -134,7 +134,7 @@ Name | Type | Description
 **defsBucketAccessLogsConfig**? | <code>[ServerlessClamscanLoggingProps](#cdk-serverless-clamscan-serverlessclamscanloggingprops)</code> | Whether or not to enable Access Logging for the Virus Definitions bucket, you can specify an existing bucket and prefix (Default: Creates a new S3 Bucket for access logs ).<br/>__*Optional*__
 **efsEncryption**? | <code>boolean</code> | Whether or not to enable encryption on EFS filesystem (Default: enabled).<br/>__*Optional*__
 **onError**? | <code>[IDestination](#aws-cdk-aws-lambda-idestination)</code> | The Lambda Destination for files that fail to scan and are marked 'ERROR' or stuck 'IN PROGRESS' due to a Lambda timeout (Default: Creates and publishes to a new SQS queue if unspecified).<br/>__*Optional*__
-**onResult**? | <code>[IDestination](#aws-cdk-aws-lambda-idestination)</code> | The Lambda Destination for files marked 'CLEAN' or 'INFECTED' based on the ClamAV Virus scan (Default: Creates and publishes to a new Event Bridge Bus if unspecified).<br/>__*Optional*__
+**onResult**? | <code>[IDestination](#aws-cdk-aws-lambda-idestination)</code> | The Lambda Destination for files marked 'CLEAN' or 'INFECTED' based on the ClamAV Virus scan or 'N/A' for scans triggered by S3 folder creation events marked (Default: Creates and publishes to a new Event Bridge Bus if unspecified).<br/>__*Optional*__
 
 
 

--- a/assets/lambda/code/scan/Dockerfile
+++ b/assets/lambda/code/scan/Dockerfile
@@ -7,7 +7,7 @@ LABEL version=1.0
 
 RUN yum install -y amazon-linux-extras \
     && PYTHON=python2 amazon-linux-extras install epel \
-    && yum -y install clamav clamd p7zip-full \
+    && yum -y install clamav clamd p7zip \
     && yum clean all \
     && pip3 install --no-cache-dir cffi awslambdaric boto3 requests aws-lambda-powertools \
     && ln -s /etc/freshclam.conf /tmp/freshclam.conf

--- a/assets/lambda/code/scan/lambda.py
+++ b/assets/lambda/code/scan/lambda.py
@@ -1,6 +1,7 @@
 # Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 # SPDX-License-Identifier: Apache-2.0
 
+from posixpath import join
 import boto3
 import botocore
 import glob
@@ -23,6 +24,7 @@ INPROGRESS = "IN PROGRESS"
 CLEAN = "CLEAN"
 INFECTED = "INFECTED"
 ERROR = "ERROR"
+SKIP = "N/A"
 
 MAX_BYTES = 4000000000
 
@@ -64,19 +66,34 @@ def lambda_handler(event, context):
     bucket_info = event["Records"][0]["s3"]
     input_bucket = bucket_info["bucket"]["name"]
     input_key = unquote_plus(bucket_info["object"]["key"])
-    mount_path = os.environ["EFS_MOUNT_PATH"]
-    definitions_path = f"{mount_path}/{os.environ['EFS_DEF_PATH']}"
-    payload_path = f"{mount_path}/{context.aws_request_id}"
-    set_status(input_bucket, input_key, INPROGRESS)
-    create_dir(input_bucket, input_key, payload_path)
-    download_object(input_bucket, input_key, payload_path)
-    expand_if_large_archive(
-        input_bucket, input_key, payload_path, bucket_info["object"]["size"]
-    )
-    create_dir(input_bucket, input_key, definitions_path)
-    freshclam_update(input_bucket, input_key, payload_path, definitions_path)
-    summary = scan(input_bucket, input_key, payload_path, definitions_path)
-    delete(payload_path)
+    summary = ""
+    if not input_key.endswith("/"):
+        mount_path = os.environ["EFS_MOUNT_PATH"]
+        definitions_path = f"{mount_path}/{os.environ['EFS_DEF_PATH']}"
+        payload_path = f"{mount_path}/{context.aws_request_id}"
+        set_status(input_bucket, input_key, INPROGRESS)
+        create_dir(input_bucket, input_key, payload_path)
+        download_object(input_bucket, input_key, payload_path)
+        expand_if_large_archive(
+            input_bucket,
+            input_key,
+            payload_path,
+            bucket_info["object"]["size"],
+        )
+        create_dir(input_bucket, input_key, definitions_path)
+        freshclam_update(
+            input_bucket, input_key, payload_path, definitions_path
+        )
+        summary = scan(input_bucket, input_key, payload_path, definitions_path)
+        delete(payload_path)
+    else:
+        summary = {
+            "source": "serverless-clamscan",
+            "input_bucket": input_bucket,
+            "input_key": input_key,
+            "status": SKIP,
+            "message": "S3 Event trigger was for a non-file object",
+        }
     logger.info(summary)
     return summary
 
@@ -98,9 +115,13 @@ def set_status(bucket, key, status):
 def create_dir(input_bucket, input_key, download_path):
     """Creates a directory at the specified location
     if it does not already exists"""
-    if not os.path.exists(download_path):
+    sub_dir = os.path.dirname(input_key)
+    full_path = download_path
+    if len(sub_dir) > 0:
+        full_path = os.path.join(full_path, sub_dir)
+    if not os.path.exists(full_path):
         try:
-            os.makedirs(download_path, exist_ok=True)
+            os.makedirs(full_path, exist_ok=True)
         except OSError as e:
             report_failure(input_bucket, input_key, download_path, str(e))
 

--- a/src/index.ts
+++ b/src/index.ts
@@ -2,18 +2,47 @@
 // SPDX-License-Identifier: Apache-2.0
 
 import * as path from 'path';
-import { Vpc, SubnetType, GatewayVpcEndpoint, GatewayVpcEndpointAwsService, Port, SecurityGroup } from '@aws-cdk/aws-ec2';
+import {
+  Vpc,
+  SubnetType,
+  GatewayVpcEndpoint,
+  GatewayVpcEndpointAwsService,
+  Port,
+  SecurityGroup,
+} from '@aws-cdk/aws-ec2';
 import { FileSystem, LifecyclePolicy, PerformanceMode } from '@aws-cdk/aws-efs';
 import { EventBus, Rule, Schedule } from '@aws-cdk/aws-events';
 import { LambdaFunction } from '@aws-cdk/aws-events-targets';
-import { Effect, PolicyStatement, ArnPrincipal, AnyPrincipal, AccountRootPrincipal } from '@aws-cdk/aws-iam';
-import { DockerImageCode, DockerImageFunction, Function, IDestination, FileSystem as LambdaFileSystem, Runtime, Code } from '@aws-cdk/aws-lambda';
-import { EventBridgeDestination, SqsDestination } from '@aws-cdk/aws-lambda-destinations';
+import {
+  Effect,
+  PolicyStatement,
+  ArnPrincipal,
+  AnyPrincipal,
+  AccountRootPrincipal,
+} from '@aws-cdk/aws-iam';
+import {
+  DockerImageCode,
+  DockerImageFunction,
+  Function,
+  IDestination,
+  FileSystem as LambdaFileSystem,
+  Runtime,
+  Code,
+} from '@aws-cdk/aws-lambda';
+import {
+  EventBridgeDestination,
+  SqsDestination,
+} from '@aws-cdk/aws-lambda-destinations';
 import { S3EventSource } from '@aws-cdk/aws-lambda-event-sources';
 import { Bucket, BucketEncryption, EventType } from '@aws-cdk/aws-s3';
 import { Queue, QueueEncryption } from '@aws-cdk/aws-sqs';
-import { Construct, Duration, CustomResource, RemovalPolicy, Stack } from '@aws-cdk/core';
-
+import {
+  Construct,
+  Duration,
+  CustomResource,
+  RemovalPolicy,
+  Stack,
+} from '@aws-cdk/core';
 
 /**
  * Interface for ServerlessClamscan Virus Definitions S3 Bucket Logging.
@@ -38,7 +67,7 @@ export interface ServerlessClamscanProps {
    */
   readonly buckets?: Bucket[];
   /**
-   * The Lambda Destination for files marked 'CLEAN' or 'INFECTED' based on the ClamAV Virus scan (Default: Creates and publishes to a new Event Bridge Bus if unspecified).
+   * The Lambda Destination for files marked 'CLEAN' or 'INFECTED' based on the ClamAV Virus scan or 'N/A' for scans triggered by S3 folder creation events marked (Default: Creates and publishes to a new Event Bridge Bus if unspecified).
    */
   readonly onResult?: IDestination;
   /**
@@ -80,7 +109,7 @@ export interface ServerlessClamscanProps {
      "source": "serverless-clamscan",
      "input_bucket": <input_bucket_name>,
      "input_key": <object_key>,
-     "status": <"CLEAN"|"INFECTED">,
+     "status": <"CLEAN"|"INFECTED"|"N/A">,
      "message": <scan_summary>,
    }
   ```
@@ -93,49 +122,48 @@ export class ServerlessClamscan extends Construct {
   /**
     The Lambda Destination for failed on erred scans [ERROR, IN PROGRESS (If error is due to Lambda timeout)].
    */
-  public readonly errorDest: IDestination
+  public readonly errorDest: IDestination;
 
   /**
     The Lambda Destination for completed ClamAV scans [CLEAN, INFECTED].
    */
-  public readonly resultDest: IDestination
+  public readonly resultDest: IDestination;
 
   /**
     Conditional: The SQS Queue for erred scans if a failure (onError) destination was not specified.
    */
-  public readonly errorQueue?: Queue
+  public readonly errorQueue?: Queue;
 
   /**
     Conditional: The SQS Dead Letter Queue for the errorQueue if a failure (onError) destination was not specified.
    */
-  public readonly errorDeadLetterQueue?: Queue
+  public readonly errorDeadLetterQueue?: Queue;
 
   /**
     Conditional: The Event Bridge Bus for completed ClamAV scans if a success (onResult) destination was not specified.
    */
-  public readonly resultBus?: EventBus
+  public readonly resultBus?: EventBus;
 
   /**
     Conditional: An Event Bridge Rule for files that are marked 'CLEAN' by ClamAV if a success destination was not specified.
    */
-  public readonly cleanRule?: Rule
+  public readonly cleanRule?: Rule;
 
   /**
     Conditional: An Event Bridge Rule for files that are marked 'INFECTED' by ClamAV if a success destination was not specified.
    */
-  public readonly infectedRule?: Rule
+  public readonly infectedRule?: Rule;
 
   /**
     Conditional: The Bucket for access logs for the virus definitions bucket if logging is enabled (defsBucketAccessLogsConfig).
    */
-  public readonly defsAccessLogsBucket?: Bucket
+  public readonly defsAccessLogsBucket?: Bucket;
 
-
-  private _scanFunction: DockerImageFunction
-  private _s3Gw: GatewayVpcEndpoint
-  private _efsRootPath = '/lambda'
-  private _efsMountPath = `/mnt${this._efsRootPath}`
-  private _efsDefsPath = 'virus_database/'
+  private _scanFunction: DockerImageFunction;
+  private _s3Gw: GatewayVpcEndpoint;
+  private _efsRootPath = '/lambda';
+  private _efsMountPath = `/mnt${this._efsRootPath}`;
+  private _efsDefsPath = 'virus_database/';
 
   /**
    * Creates a ServerlessClamscan construct.
@@ -144,7 +172,6 @@ export class ServerlessClamscan extends Construct {
    * @param props A `ServerlessClamscanProps` interface.
    */
   constructor(scope: Construct, id: string, props: ServerlessClamscanProps) {
-
     super(scope, id);
 
     if (!props.onResult) {
@@ -195,10 +222,12 @@ export class ServerlessClamscan extends Construct {
     }
 
     const vpc = new Vpc(this, 'ScanVPC', {
-      subnetConfiguration: [{
-        subnetType: SubnetType.ISOLATED,
-        name: 'Isolated',
-      }],
+      subnetConfiguration: [
+        {
+          subnetType: SubnetType.ISOLATED,
+          name: 'Isolated',
+        },
+      ],
     });
 
     vpc.addFlowLog('FlowLogs');
@@ -209,7 +238,7 @@ export class ServerlessClamscan extends Construct {
 
     const fileSystem = new FileSystem(this, 'ScanFileSystem', {
       vpc: vpc,
-      encrypted: (props.efsEncryption === false) ? false : true,
+      encrypted: props.efsEncryption === false ? false : true,
       lifecyclePolicy: LifecyclePolicy.AFTER_7_DAYS,
       performanceMode: PerformanceMode.GENERAL_PURPOSE,
       removalPolicy: RemovalPolicy.DESTROY,
@@ -219,41 +248,46 @@ export class ServerlessClamscan extends Construct {
       }),
     });
 
-    const lambda_ap = fileSystem.addAccessPoint('ScanLambdaAP',
-      {
-        createAcl: {
-          ownerGid: '1000',
-          ownerUid: '1000',
-          permissions: '755',
-        },
-        posixUser: {
-          gid: '1000',
-          uid: '1000',
-        },
-        path: this._efsRootPath,
-      });
+    const lambda_ap = fileSystem.addAccessPoint('ScanLambdaAP', {
+      createAcl: {
+        ownerGid: '1000',
+        ownerUid: '1000',
+        permissions: '755',
+      },
+      posixUser: {
+        gid: '1000',
+        uid: '1000',
+      },
+      path: this._efsRootPath,
+    });
 
     const logs_bucket = props.defsBucketAccessLogsConfig?.logsBucket;
     const logs_bucket_prefix = props.defsBucketAccessLogsConfig?.logsPrefix;
     if (logs_bucket === true || logs_bucket === undefined) {
-      this.defsAccessLogsBucket = new Bucket(this, 'VirusDefsAccessLogsBucket', {
-        encryption: BucketEncryption.S3_MANAGED,
-        removalPolicy: RemovalPolicy.RETAIN,
-      });
-      this.defsAccessLogsBucket.addToResourcePolicy(new PolicyStatement({
-        effect: Effect.DENY,
-        actions: ['s3:*'],
-        resources: [
-          this.defsAccessLogsBucket.arnForObjects('*'),
-          this.defsAccessLogsBucket.bucketArn,
-        ],
-        principals: [new AnyPrincipal()],
-        conditions: {
-          Bool: {
-            'aws:SecureTransport': false,
-          },
+      this.defsAccessLogsBucket = new Bucket(
+        this,
+        'VirusDefsAccessLogsBucket',
+        {
+          encryption: BucketEncryption.S3_MANAGED,
+          removalPolicy: RemovalPolicy.RETAIN,
         },
-      }));
+      );
+      this.defsAccessLogsBucket.addToResourcePolicy(
+        new PolicyStatement({
+          effect: Effect.DENY,
+          actions: ['s3:*'],
+          resources: [
+            this.defsAccessLogsBucket.arnForObjects('*'),
+            this.defsAccessLogsBucket.bucketArn,
+          ],
+          principals: [new AnyPrincipal()],
+          conditions: {
+            Bool: {
+              'aws:SecureTransport': false,
+            },
+          },
+        }),
+      );
     } else if (logs_bucket != false) {
       this.defsAccessLogsBucket = logs_bucket;
     }
@@ -263,56 +297,63 @@ export class ServerlessClamscan extends Construct {
       removalPolicy: RemovalPolicy.DESTROY,
       autoDeleteObjects: true,
       serverAccessLogsBucket: this.defsAccessLogsBucket,
-      serverAccessLogsPrefix: logs_bucket === false ? undefined: logs_bucket_prefix,
+      serverAccessLogsPrefix:
+        logs_bucket === false ? undefined : logs_bucket_prefix,
     });
 
-
-    defs_bucket.addToResourcePolicy(new PolicyStatement({
-      effect: Effect.DENY,
-      actions: ['s3:*'],
-      resources: [
-        defs_bucket.arnForObjects('*'),
-        defs_bucket.bucketArn,
-      ],
-      principals: [new AnyPrincipal()],
-      conditions: {
-        Bool: {
-          'aws:SecureTransport': false,
+    defs_bucket.addToResourcePolicy(
+      new PolicyStatement({
+        effect: Effect.DENY,
+        actions: ['s3:*'],
+        resources: [defs_bucket.arnForObjects('*'), defs_bucket.bucketArn],
+        principals: [new AnyPrincipal()],
+        conditions: {
+          Bool: {
+            'aws:SecureTransport': false,
+          },
         },
-      },
-    }),
+      }),
     );
-    defs_bucket.addToResourcePolicy(new PolicyStatement({
-      effect: Effect.ALLOW,
-      actions: ['s3:GetObject', 's3:ListBucket'],
-      resources: [defs_bucket.arnForObjects('*'), defs_bucket.bucketArn],
-      principals: [new AnyPrincipal()],
-      conditions: {
-        StringEquals: {
-          'aws:SourceVpce': this._s3Gw.vpcEndpointId,
+    defs_bucket.addToResourcePolicy(
+      new PolicyStatement({
+        effect: Effect.ALLOW,
+        actions: ['s3:GetObject', 's3:ListBucket'],
+        resources: [defs_bucket.arnForObjects('*'), defs_bucket.bucketArn],
+        principals: [new AnyPrincipal()],
+        conditions: {
+          StringEquals: {
+            'aws:SourceVpce': this._s3Gw.vpcEndpointId,
+          },
         },
-      },
-    }),
+      }),
     );
-    defs_bucket.addToResourcePolicy(new PolicyStatement({
-      effect: Effect.DENY,
-      actions: ['s3:PutBucketPolicy', 's3:DeleteBucketPolicy'],
-      resources: [defs_bucket.bucketArn],
-      notPrincipals: [new AccountRootPrincipal()],
-    }),
+    defs_bucket.addToResourcePolicy(
+      new PolicyStatement({
+        effect: Effect.DENY,
+        actions: ['s3:PutBucketPolicy', 's3:DeleteBucketPolicy'],
+        resources: [defs_bucket.bucketArn],
+        notPrincipals: [new AccountRootPrincipal()],
+      }),
     );
-    this._s3Gw.addToPolicy(new PolicyStatement({
-      effect: Effect.ALLOW,
-      actions: ['s3:GetObject', 's3:ListBucket'],
-      resources: [defs_bucket.arnForObjects('*'), defs_bucket.bucketArn],
-      principals: [new AnyPrincipal()],
-    }));
+    this._s3Gw.addToPolicy(
+      new PolicyStatement({
+        effect: Effect.ALLOW,
+        actions: ['s3:GetObject', 's3:ListBucket'],
+        resources: [defs_bucket.arnForObjects('*'), defs_bucket.bucketArn],
+        principals: [new AnyPrincipal()],
+      }),
+    );
 
     this._scanFunction = new DockerImageFunction(this, 'ServerlessClamscan', {
-      code: DockerImageCode.fromImageAsset(path.join(__dirname, '../assets/lambda/code/scan')),
+      code: DockerImageCode.fromImageAsset(
+        path.join(__dirname, '../assets/lambda/code/scan'),
+      ),
       onSuccess: this.resultDest,
       onFailure: this.errorDest,
-      filesystem: LambdaFileSystem.fromEfsAccessPoint(lambda_ap, this._efsMountPath),
+      filesystem: LambdaFileSystem.fromEfsAccessPoint(
+        lambda_ap,
+        this._efsMountPath,
+      ),
       vpc: vpc,
       vpcSubnets: { subnets: vpc.isolatedSubnets },
       allowAllOutbound: false,
@@ -326,11 +367,16 @@ export class ServerlessClamscan extends Construct {
         POWERTOOLS_SERVICE_NAME: 'virus-scan',
       },
     });
-    this._scanFunction.connections.allowToAnyIpv4(Port.tcp(443), 'Allow outbound HTTPS traffic for S3 access.');
+    this._scanFunction.connections.allowToAnyIpv4(
+      Port.tcp(443),
+      'Allow outbound HTTPS traffic for S3 access.',
+    );
     defs_bucket.grantRead(this._scanFunction);
 
     const download_defs = new DockerImageFunction(this, 'DownloadDefs', {
-      code: DockerImageCode.fromImageAsset(path.join(__dirname, '../assets/lambda/code/download_defs')),
+      code: DockerImageCode.fromImageAsset(
+        path.join(__dirname, '../assets/lambda/code/download_defs'),
+      ),
       timeout: Duration.minutes(5),
       memorySize: 1024,
       environment: {
@@ -342,15 +388,17 @@ export class ServerlessClamscan extends Construct {
 
     if (download_defs.role) {
       const download_defs_role = `arn:${stack.partition}:sts::${stack.account}:assumed-role/${download_defs.role.roleName}/${download_defs.functionName}`;
-      const download_defs_assumed_principal = new ArnPrincipal(download_defs_role);
-      defs_bucket.addToResourcePolicy(new PolicyStatement({
-        effect: Effect.DENY,
-        actions: ['s3:PutObject*'],
-        resources: [
-          defs_bucket.arnForObjects('*'),
-        ],
-        notPrincipals: [download_defs.role, download_defs_assumed_principal],
-      }));
+      const download_defs_assumed_principal = new ArnPrincipal(
+        download_defs_role,
+      );
+      defs_bucket.addToResourcePolicy(
+        new PolicyStatement({
+          effect: Effect.DENY,
+          actions: ['s3:PutObject*'],
+          resources: [defs_bucket.arnForObjects('*')],
+          notPrincipals: [download_defs.role, download_defs_assumed_principal],
+        }),
+      );
     }
 
     defs_bucket.grantReadWrite(download_defs);
@@ -362,7 +410,9 @@ export class ServerlessClamscan extends Construct {
 
     const init_defs_cr = new Function(this, 'InitDefs', {
       runtime: Runtime.PYTHON_3_8,
-      code: Code.fromAsset(path.join(__dirname, '../assets/lambda/code/initialize_defs_cr')),
+      code: Code.fromAsset(
+        path.join(__dirname, '../assets/lambda/code/initialize_defs_cr'),
+      ),
       handler: 'lambda.lambda_handler',
       timeout: Duration.minutes(5),
     });
@@ -373,11 +423,10 @@ export class ServerlessClamscan extends Construct {
       properties: {
         FnName: download_defs.functionName,
       },
-    },
-    );
+    });
 
     if (props.buckets) {
-      props.buckets.forEach(bucket => {
+      props.buckets.forEach((bucket) => {
         this.addSourceBucket(bucket);
       });
     }
@@ -390,43 +439,57 @@ export class ServerlessClamscan extends Construct {
    * @param bucket The bucket to add the scanning bucket policy and s3:ObjectCreate* trigger to.
    */
   addSourceBucket(bucket: Bucket) {
-    this._scanFunction.addEventSource(new S3EventSource(bucket, { events: [EventType.OBJECT_CREATED] }));
+    this._scanFunction.addEventSource(
+      new S3EventSource(bucket, { events: [EventType.OBJECT_CREATED] }),
+    );
     bucket.grantRead(this._scanFunction);
-    this._scanFunction.addToRolePolicy(new PolicyStatement({
-      effect: Effect.ALLOW,
-      actions: ['s3:PutObjectTagging', 's3:PutObjectVersionTagging'],
-      resources: [bucket.arnForObjects('*')],
-    }));
+    this._scanFunction.addToRolePolicy(
+      new PolicyStatement({
+        effect: Effect.ALLOW,
+        actions: ['s3:PutObjectTagging', 's3:PutObjectVersionTagging'],
+        resources: [bucket.arnForObjects('*')],
+      }),
+    );
 
     if (this._scanFunction.role) {
       const stack = Stack.of(this);
       const scan_assumed_role = `arn:${stack.partition}:sts::${stack.account}:assumed-role/${this._scanFunction.role.roleName}/${this._scanFunction.functionName}`;
       const scan_assumed_principal = new ArnPrincipal(scan_assumed_role);
-      this._s3Gw.addToPolicy(new PolicyStatement({
-        effect: Effect.ALLOW,
-        actions: ['s3:GetObject*', 's3:GetBucket*', 's3:List*'],
-        resources: [bucket.bucketArn, bucket.arnForObjects('*')],
-        principals: [this._scanFunction.role, scan_assumed_principal],
-      }));
-      this._s3Gw.addToPolicy(new PolicyStatement({
-        effect: Effect.ALLOW,
-        actions: ['s3:PutObjectTagging', 's3:PutObjectVersionTagging'],
-        resources: [bucket.arnForObjects('*')],
-        principals: [this._scanFunction.role, scan_assumed_principal],
-      }));
+      this._s3Gw.addToPolicy(
+        new PolicyStatement({
+          effect: Effect.ALLOW,
+          actions: ['s3:GetObject*', 's3:GetBucket*', 's3:List*'],
+          resources: [bucket.bucketArn, bucket.arnForObjects('*')],
+          principals: [this._scanFunction.role, scan_assumed_principal],
+        }),
+      );
+      this._s3Gw.addToPolicy(
+        new PolicyStatement({
+          effect: Effect.ALLOW,
+          actions: ['s3:PutObjectTagging', 's3:PutObjectVersionTagging'],
+          resources: [bucket.arnForObjects('*')],
+          principals: [this._scanFunction.role, scan_assumed_principal],
+        }),
+      );
 
       // Need the assumed role for the not Principal Action with Lambda
-      bucket.addToResourcePolicy(new PolicyStatement({
-        effect: Effect.DENY,
-        actions: ['s3:GetObject'],
-        resources: [bucket.arnForObjects('*')],
-        notPrincipals: [this._scanFunction.role, scan_assumed_principal],
-        conditions: {
-          StringEquals: {
-            's3:ExistingObjectTag/scan-status': ['IN PROGRESS', 'INFECTED', 'ERROR'],
+      bucket.addToResourcePolicy(
+        new PolicyStatement({
+          effect: Effect.DENY,
+          actions: ['s3:GetObject'],
+          resources: [bucket.arnForObjects('*')],
+          notPrincipals: [this._scanFunction.role, scan_assumed_principal],
+          conditions: {
+            StringEquals: {
+              's3:ExistingObjectTag/scan-status': [
+                'IN PROGRESS',
+                'INFECTED',
+                'ERROR',
+              ],
+            },
           },
-        },
-      }));
+        }),
+      );
     }
   }
 }

